### PR TITLE
Fixing unchecked error

### PR DESCRIPTION
**Pull request by saromanov**
Hi! I've just fixed unchecked error. Please check it

### CODE REVIEW

File: discovery/azure/azure.go [hunk 1]
```diff
@@ -552,7 +552,7 @@ func (client *azureClient) getNetworkInterfaceByID(networkInterfaceID string) (n
 
 	result, err = client.nic.GetResponder(resp)
 	if err != nil {
-		err = autorest.NewErrorWithError(err, "network.InterfacesClient", "Get", resp, "Failure responding to request")
+		return result, autorest.NewErrorWithError(err, "network.InterfacesClient", "Get", resp, "Failure responding to request")
```
Comment (aixeshunter) [inline discovery/azure/azure.go]: I think it's unnecessary.

Comment (gouthamve) [reply discovery/azure/azure.go]: Hmm, can you explain why? This looks like a valid change to make and if nothing, we should atleast document why the error is being ignored.

From what I see, when it errors, the result will be empty.

Comment (cstyan) [reply discovery/azure/azure.go]: Agreed, looks to me like this is an error that is slipping through the cracks right now. Might be nicer to declare `var err error` at the top like `result` and keep the `err = autorest...` line as is but change the return at the end of the function to `return result, err`?

**Comment by simonpasquier** [comment]
This is a duplicate of #4719

File: discovery/azure/azure.go (hunk 1 above)
Comment (pytimer) [reply discovery/azure/azure.go]: I see the code that invoke this method and this result unused if err is not nil. So i think is it fixes `return result,...` to `return nil, autorest.NewErrorWithError()` will be better?

**Comment by jannickfahlbusch** [comment]
#4719 just got merged. You can close this PR. 

Thanks for your effort :)

**Pull request closed without merging**